### ISSUE TITLE
Filtering strategies for discrepant phenotypes

### ISSUE THREAD

**Issue by mbhall88** [mbhall88/drprg#22]
I think it might be wise for use to use some kind of phenotype filtering strategy.

https://github.com/mbhall88/drprg/issues/21#issuecomment-1352351317 is a prime example of why.

In the NEJM Cryptic paper I know they ignored any samples where 

> Laboratory error was assumed in instances in which three or more phenotypes were discordant with the genotype of an isolate or in which susceptible phenotypes were recorded despite the presence of the high-level resistance mutation katG S315T for isoniazid or rpoB S450L for rifampin.[14](https://www.nejm.org/doi/10.1056/NEJMoa1800474#) Such isolates were excluded from further analysis.

I wonder if we do something similar?

**Comment by iqbal-lab** [comment]
I totally agree. Also basic wgs filtering- is there at least 90% of the genome having >20x depth or something

**Comment by mbhall88** [comment]
I've got a min. 15x depth filter on at the moment. Only two isolates have more than 1 panel gene deleted, so I could just manually remove those.

**Comment by mbhall88** [comment]
Should we do 3 or more discordant phenotypes on all tools, or any tool, or drprg?

**Comment by iqbal-lab** [comment]
my real concern with how to do this, is that we can't filter samples based on discordant phenotypes where a legit explanation for the discordance is the catalogue is limited. 
So where the discordance == false negatives, i'm not keen.
But for discordances where there is a solid catalogue mutation present, but phenotype==S, and where that happens for multiple drugs, thats a defensible approach. the problem is how to define "solid catalogue gene". Int he NEJM paper they took just katG S315T and rpoB S450L .

**Comment by mbhall88** [comment]
Yeah, good point. Fps only seems the wisest option.

I have noticed a decent number of emb M306 FPs as well. So maybe we need to look at the WHO catalogue and get a list of the most common mutations and base it on that?

**Comment by iqbal-lab** [comment]
AH. Emb306 is another story.

**Comment by lachlancoin** [comment]
This seems like a bit of a slippery slope to do phenotype filtering based
on discordance with genotype

**Comment by iqbal-lab** [comment]
It is  , I think its only justifiable as a route to arguing there's a sample switch/error, if you see multiple v unlikely discrepancies in one sample. But I'm hoping it becomes unnecessary if we reject non wgs samples

**Comment by mbhall88** [comment]
We can always report two values - filtered and unfiltered. And filtered could just be matching the NEJM paper. That way we can compare Sn and Sp to the NEJM results, but also the unfiltered so it doesn't look like we're hiding anything

**Comment by mbhall88** [comment]
Okay, so if we do filtering based on pheno-geno disagreement, it should *only* be based on FPs as FNs could just be catalogue gaps.

For the current dataset, we have the following

- 1394 runs have FP for 1 drug(s), on all tools  - 0 are FP for RIF and INH
- 381 runs have FP for 2 drug(s), on all tools  - 16 are FP for RIF and INH
- 52 runs have FP for 3 drug(s), on all tools  - 4 are FP for RIF and INH
- 9 runs have FP for 4 drug(s), on all tools  - 2 are FP for RIF and INH
- 1 runs have FP for 5 drug(s), on all tools  - 1 are FP for RIF and INH

I am also collapsing fluoroquinolones and aminoglycosides so that we don't double-count things like FP on levofloxacin and moxifloxacin and ofloxacin.

This seems to be the most fair way to filter as we don't favour any tools; we require that a run is FP on **all** tools.

Do we want to do this kind of filtering? Or just leave it all unfiltered?

**Comment by mbhall88** [comment]
This recent Lancet Microbe Correspondence is on interest to this discussion https://doi.org/10.1016/S2666-5247(23)00007-1

**Comment by lachlancoin** [comment]
If these are genuine catalog gaps, then each tool is penalised in the same
way .  So the only point is to make the sensitivity and specificity better
for everything, and I guess I dont really see the point of that.

**Comment by lachlancoin** [comment]
I can see why you might want to do this in the context of reporting a
catalogue though, when you really do care about how much you are missing ..
but I dont think its appropriate  in the context of benchmarking.

**Comment by lachlancoin** [comment]
Ah, that makes sense why people would do that kind of filtering to meet the
WHO product profiles.  Can be discussed as a reason why our sensitivities
are lower than those reported in catalog
